src/Mvc/src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Binders/ComplexTypeModelBinder.cs

```diff
@@ -173,8 +180,19 @@ private async Task BindModelCoreAsync(ModelBindingContext bindingContext)
                 bindingContext.ModelState.TryAddModelError(bindingContext.ModelName, message);
             }
 
-            bindingContext.Result = ModelBindingResult.Success(bindingContext.Model);
             _logger.DoneAttemptingToBindModel(bindingContext);
+
+            // Have all binders failed because no data was available? Idea is to continue if for data was available but
+            // the binders failed due to (say) conversion errors.
```

Comment:
It would be great to see an example of the specific case we discussed: collection of (complex type with (string, IFormFile)) and explain how it won't interrupt binding the collection if/when you have a conversion error.